### PR TITLE
port parabricks/mutectcaller to nf-test

### PR DESCRIPTION
**Pull request by famosab**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by famosab** [comment]
PON test does not work because:
    // // PON test not passing because the tool expects all chromosomes in the PON file to
    // // be present in the reference genome.
    
Tumor-Normal tests do not work because:
    // // tumor-normal calling not passing because of issue with readgroups in the test files
    // // [PB Error][src/readers.cpp:1314] ID simulation01 maps to tm_simulation01, but PU is nm_simulation01 (different platform), exiting.

File: modules/nf-core/parabricks/mutectcaller/main.nf [hunk 1]
```diff
@@ -1,8 +1,9 @@
 process PARABRICKS_MUTECTCALLER {
     tag "$meta.id"
     label 'process_high'
+    label 'process_gpu'
```
Comment (famosab) [inline modules/nf-core/parabricks/mutectcaller/main.nf:5]:
```suggestion
    label 'process_gpu'
    stageInMode 'copy' // needed by the module to work properly - can be removed once fixed upstream

```

File: modules/nf-core/parabricks/mutectcaller/tests/nextflow.config [hunk 2]
```diff
@@ -0,0 +1,8 @@
+process {
+
+    withName: PARABRICKS_MUTECTCALLER {
+        ext.args = params.module_args
+        stageInMode = 'copy'
```
Comment (famosab) [inline modules/nf-core/parabricks/mutectcaller/tests/nextflow.config:5]:
```suggestion
```

File: modules/nf-core/parabricks/mutectcaller/main.nf [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+    stageInMode 'copy' // needed by the module to work properly - can be removed once fixed upstream
```
Comment (SPPearce) [inline modules/nf-core/parabricks/mutectcaller/main.nf:5]: Do you have a PR that could be linked to? Then it makes it easy to see whether it has been addressed.

File: modules/nf-core/parabricks/mutectcaller/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        when {
+            params {
+                module_args = "--mutect-low-memory"
+            }
+            process {
+                """
+                input[0] = [
+                    [ id:'test',  tumor_id:'tumour' ],
+                    file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
+                    [], // index only neeeded if using intervals or tumor-normal calling
```
Comment (SPPearce) [inline modules/nf-core/parabricks/mutectcaller/tests/main.nf.test:26]:
```suggestion
                    [], // index only needed if using intervals or tumor-normal calling
```

**Review by SPPearce — COMMENTED** [review]
Looks good to me. Can you just link to an issue to help track the stageInMode please, and make sure there is an issue to fix the tests for tumour/normal mode.

**Comment by famosab** [comment]
@SPPearce Do you think its fine to leave in the outcommented tests? I left them for later reference or that someone whos a bit more skilled might be able to fix them - but the general testing works so I did not bother to spend more time on this.

**Comment by famosab** [comment]
Opened an issue in this repo, will link if I find any description at the parabricks website: https://github.com/nf-core/modules/issues/7226

**Comment by SPPearce** [comment]
> @SPPearce Do you think its fine to leave in the outcommented tests? I left them for later reference or that someone whos a bit more skilled might be able to fix them - but the general testing works so I did not bother to spend more time on this.

Fine by me to leave them.

**Review by SPPearce — APPROVED** [review]

**Pull request merged**